plugins/frontend/backstage-plugin-github-pull-requests/src/components/useGithubLoggedIn.tsx

```diff
@@ -37,6 +37,7 @@ export const useGithubLoggedIn = () => {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
 
   useEffect(() => {
+    githubApi.getAccessToken('repo', { optional: true });
```

Comment:
Thanks so much for this. :) At the moment I am not fully understanding the change.

Why do we need to retrieve the access token here?
Should we await for the token from the promise?
Should the option also include: `instantPopup: false` so that it does not prompt the user?

As I understand it, we do not want to prompt the user to login but simply tell if the user is loggedin in the component..

Reply:
@punkle here's some more context:

The issue is that `githubApi.sessionState$()` reflects a non signed-in state if nothing has triggered the auth flow yet. This change is meant to optionally trigger the auth flow and check if the user can be considered logged in if a cached access token exists. The result of the request for the access token here does not really matter and we don't need to await it, we just use it to check if this access is already granted which will update the `githubApi.sessionState$()` observable if so. `instantPopup: false` is not necessary since it defaults to `false`. This will not prompt the user. See relevant docs for these options [here](https://github.com/backstage/backstage/blob/d06fbf7bb0d182aab04e70e325e483f5b9f41038/packages/core-plugin-api/src/apis/definitions/auth.ts#L78-L97).

This is the same pattern that Backstage's [`SignInPage`](https://github.com/backstage/backstage/blob/0c31fdb95848211de355ef954529515cd1100342/packages/core-components/src/layout/SignInPage/SignInPage.tsx#L111-L114) performs to check for an existing session so that you're not prompted to re-login everytime the app is refreshed

Reply:
Thank you for the explanation! :)

Reply:
@punkle any other concerns here?

Reply:
No concerns. Thanks for the explanation.